airflow/providers/pinecone/hooks/pinecone.py

```diff
@@ -135,119 +170,106 @@ def upsert(
             **kwargs,
         )
 
-    @staticmethod
+    def get_pod_spec_obj(
+        self,
+        replicas: int | None = None,
+        shards: int | None = None,
+        pods: int | None = None,
+        pod_type: str | None = "p1.x1",
+        metadata_config: dict | None = None,
+        source_collection: str | None = None,
+        environment: str | None = None,
+    ) -> PodSpec:
+        return PodSpec(
+            environment=environment or self.environment,
+            replicas=replicas,
+            shards=shards,
+            pods=pods,
+            pod_type=pod_type,
+            metadata_config=metadata_config,
+            source_collection=source_collection,
+        )
+
+    def get_serverless_spec_obj(self, cloud, region: str | None = None) -> ServerlessSpec:
+        return ServerlessSpec(cloud=cloud, region=region or self.region)
+
     def create_index(
+        self,
```

Comment:
What about other parameters which was there previously? Is it not required?

Reply:
@sunank200 , All the other parameters are now part of the `Spec` object. Hence, removed them.

Reply:
Can you document all the parameters supported as part of specs or provide the link to pinecone docs for the same in the doc-string?

Reply:
@rawwar

Reply:
Sorry, we have separate methods for Spec objects and their docstrings talk about all the required parameters. Those two functions are right above this method. I think I'll update the docstring to mention about using the relevant methods to generate the Spec object. I guess that would clear things